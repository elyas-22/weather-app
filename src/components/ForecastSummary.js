import React from "react";

function ForecastSummary(props) {
  const { date, description, temperature, icon } = props;
  const longdate = new Date(date);
  const newdate = longdate.toDateString(0);
  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary_date">{newdate}</div>
      <div className="forecast-summary_description">{description}</div>
      <div className="forecast-summary_icon" data-testid="forecast-icon">
        {icon}
      </div>
      <div className="forecast-summary_temperature">
        {temperature.max}&#8451;
      </div>
    </div>
  );
}

/* function ForecastSummary(props) {
  const { date, description, temperature, icon } = props;
  const longdate = new Date(date);
  const newdate = longdate.toDateString(0);
  return (
    <div className="forecast-summary">
      <div className="forecast-summary_date">{newdate}</div>
      <div className="forecast-summary_description">{description}</div>
      <div className="forecast-summary_temperature">
        {temperature.max} &#8451;
      </div>
      <div className="forecast-summary_icon">{icon}</div>
    </div>
  );
} */

export default ForecastSummary;
