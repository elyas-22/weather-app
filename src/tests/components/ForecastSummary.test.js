import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub Description",
    icon: "forecast-Icon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("Renders ok", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for the props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(getByText("1111111")).toHaveClass("forecast-summary_date");
    expect(getByText("Stub Description")).toHaveClass(
      "forecast-summary_description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary_icon");
    expect(getByText("22â„ƒ")).toHaveClass("forecast-summary_temperature");
  });
});
