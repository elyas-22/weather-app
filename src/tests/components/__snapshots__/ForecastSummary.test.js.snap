// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary Renders ok 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    >
      1111111
    </div>
    <div
      class="forecast-summary_description"
    >
      Stub Description
    </div>
    <div
      class="forecast-summary_icon"
      data-testid="forecast-icon"
    >
      forecast-Icon
    </div>
    <div
      class="forecast-summary_temperature"
    >
      22â„ƒ
    </div>
  </div>
</DocumentFragment>
`;
